### PR TITLE
Numerous upgrades including perf and output saving.

### PR DESCRIPTION
**Pull request by jonathanhhb**
This will only work with dtk_generic_intrahost 0.1.7 (has get_mcw func).

### CODE REVIEW

File: dtk_pymod_core.py [hunk 1]
```diff
@@ -18,10 +23,10 @@ def setup_vd_callbacks():
 
 def conceive_baby_callback( individual_id, duration ):
     print( "{0} just got pregnant".format( individual_id ) )
-    gi.initiate_pregnancy( individual_id );
+    gi.initiate_pregnancy( individual_id )
 
 def update_pregnancy_callback( individual_id, dt ):
-    return gi.update_pregnancy( individual_id, int(dt) );
+    return gi.update_pregnancy( individual_id, int(dt) )
```
Comment (clorton) [inline dtk_pymod_core.py]: I guess this is okay - wondering why you are bothering with vital dynamics for this work.

File: dtk_pymod_core.py [hunk 2]
```diff
@@ -47,20 +62,43 @@ def do_shedding_update( human_pop ):
...
+def save_output( tag ): #, susceptible, exposeds, prevalence, recovered, disdeaths ):
+    filename = os.path.join(os.getcwd(), "output", tag + datetime.datetime.now().strftime('%Y-%m-%d_%H-%M-%S.csv'))
+    if not os.path.exists( "output" ):
+        os.mkdir( "output" )
+    datadict = {}
+    datadict["S"] = susceptible
+    datadict["E"] = exposeds
+    datadict["I"] = prevalence
+    datadict["R"] = recovered
+    datadict["D"] = disdeaths
```
Comment (clorton) [inline dtk_pymod_core.py]: You can do this more succinctly with the following:
```python
datadict = { "S": susceptible, "E": exposeds, "I": prevalence, "R": recovered, "D": disdeaths }
```

File: dtk_pymod_core.py [hunk 3]
```diff
@@ -47,20 +62,43 @@ def do_shedding_update( human_pop ):
...
+    datadict["S"] = susceptible
+    datadict["E"] = exposeds
+    datadict["I"] = prevalence
+    datadict["R"] = recovered
+    datadict["D"] = disdeaths 
+    with open( filename, 'w' ) as datafile: 
+        mywriter = csv.DictWriter( datafile,fieldnames=["S","E","I","R","D"] )
+        mywriter.writeheader()
+        for idx in range(len(datadict["S"])):
+            mywriter.writerow( {'S': datadict['S'][idx], 'E': datadict["E"][idx], 'I': datadict["I"][idx], 'R': datadict["R"][idx], 'D': datadict["D"][idx]})
```
Comment (clorton) [inline dtk_pymod_core.py]: If you are willing to take a pandas dependency (I think that is reasonable) you can collapse this to the following:
```python
df = pandas.DataFrame(datadict)
df.to_csv(filename, index=False)
```

File: dtk_pymod_core.py [hunk 4]
```diff
@@ -69,25 +107,28 @@ def do_vitaldynamics_update( human_pop, graveyard, census_cb = None, death_cb =
...
+        num_people += mcw
+        global fertility
+        if fertility:
+            ipm = gi.is_possible_mother( hum_id )
+            ip = gi.is_pregnant( hum_id )
+            if hum_id == 0:
+                pdb.set_trace()
+            age = gi.get_age( hum_id )
+            #print( "Calling cfp with {0}, {1}, {2}, and {3}.".format( str(ipm), str(ip), str(age), str(hum_id) ) )
+            nd.consider_for_pregnancy( ( ipm, ip, hum_id, age, 1.0 ) )
```
Comment (clorton) [inline dtk_pymod_core.py]: Why not save the call to `consider_for_pregnancy` is `ip == True` and, optionally, `age` is out of range?

File: transmission_demo.py [hunk 5]
```diff
@@ -283,26 +283,28 @@ def run( from_script = False ):
...
-        do_shedding_update( human_pop )
+        stat_pop = do_shedding_update( human_pop )
         #contagion_bucket_homog /= len(human_pop)
-        do_vitaldynamics_update( human_pop, graveyard ) 
+        do_vitaldynamics_update( human_pop, graveyard, contagion_buckets ) 
         distribute_interventions( t )
 
         #report_channels[ "Infected" ].append( prevalence[-1] )
-        print( "At end of timestep {0} num_infected = {1} stat_pop = {2}, disease deaths = {3}.".format( t, prevalence[-1], len(human_pop), len(graveyard) ) )
+        print( "At end of timestep {0} num_infected = {1} stat_pop = {2}, disease deaths = {3}.".format( t, prevalence[-1], stat_pop, len(graveyard) ) )
```
Comment (clorton) [inline transmission_demo.py]: Time for the future (Python 3.6+) - `f-strings`!
```python
print( f"At end of timestep {t} num_infected = {prevalence[-1]} stat_pop = {stat_pop}, disease deaths = {len(graveyard)}." )
```

**Comment by jonathanhhb** [comment]
Thanks for the review! 

It's only letting me respond here instead of inline... :(

Thing 1) Ya, I'm kind of a vertical list guy when it comes to readability.
Thing 2) I will save the value as suggested thanks. I'm sure it was like that due to iterative development.
Thing 3) Ya, I was totally trying to avoid an unnecessary pandas dependency. 
Thing 4) I can do the f-strings if you think that makes it look hip-per. :)

**Pull request merged**